lib/Controller/ControllerFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Controller;
+
+use InvalidArgumentException;
+use function get_class;
+use function sprintf;
+
+class ControllerFactory
```

Comment:
This class is less a factory but more a provider of already created instances. Even when only one of them is needed.